core/src/main/scala/org/http4s/ContextRoutes.scala

```diff
@@ -0,0 +1,44 @@
+package org.http4s
+
+import cats.data.{Kleisli, OptionT}
+import cats.Applicative
+import cats.effect.Sync
+
+object ContextRoutes {
+
+  /** Lifts a function into an [[ContextRoutes]].  The application of `run`
+    * is suspended in `F` to permit more efficient combination of
+    * routes via `SemigroupK`.
+    *
+    * @tparam F the effect of the [[ContextRoutes]]
+    * @tparam T the type of the auth info in the [[ContextRequest]] accepted by the [[ContextRoutes]]
+    * @param run the function to lift
+    * @return an [[ContextRoutes]] that wraps `run`
+    */
+  def apply[T, F[_]](run: ContextRequest[F, T] => OptionT[F, Response[F]])(
+      implicit F: Sync[F]): ContextRoutes[T, F] =
+    Kleisli(req => OptionT(F.suspend(run(req).value)))
+
+  /** Lifts a partial function into an [[AuthedRoutes]].  The application of the
```

Comment:
There's a couple of `AuthedRoutes` left in the comments.

Reply:
well spotted. will fix